sw/host/opentitanlib/src/transport/cw310/uart.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transport::TransportError;
+use anyhow::{ensure, Result};
+use serialport::{SerialPort, SerialPortType};
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::time::Duration;
+
+use crate::io::uart::Uart;
+use crate::transport::cw310::usb::Backend;
+
+pub struct CW310Uart {
+    port: RefCell<Box<dyn SerialPort>>,
+}
+
+impl CW310Uart {
+    pub fn open(backend: Rc<RefCell<Backend>>, instance: u32) -> Result<Self> {
+        let usb = backend.borrow();
+        let serial_number = usb.get_serial_number();
+
+        let mut ports = serialport::available_ports()?;
+        ports.retain(|port| {
+            if let SerialPortType::UsbPort(info) = &port.port_type {
+                if info.serial_number.as_deref() == Some(serial_number) {
+                    return true;
+                }
+            }
+            false
```

Comment:
I probably wouldn't, but you _could_ write

```rust
matches!(&port.port_type, SerialPortType::UsbPort(info) if info.serial_number.as_deref() = Some(serial_number))
```

Reply:
I probably wouldn't either.